boards/arm/colibri_imx7d_m4/doc/colibri_imx7d_m4.rst

```diff
@@ -0,0 +1,339 @@
++------------+-----------------------+------------------------+-----------------------+----------------------+
+| TCML       | 0x007F8000-0x007FFFFF |                        | 0x1FFF8000-0x1FFFFFFF | 32KB                 |
++------------+-----------------------+------------------------+-----------------------+----------------------+
+| OCRAM_S    | 0x00180000-0x00187FFF | 0x20180000-0x20187FFF  | 0x00000000-0x00007FFF | 32KB                 |
++------------+-----------------------+------------------------+-----------------------+----------------------+
+| QSPI Flash |                       |                        | 0x08000000-0x0BFFFFFF | 64MB                 |
++------------+-----------------------+------------------------+-----------------------+----------------------+
+
+
+References
+==========
+
+- `i.MX 7 Dual Reference Manual`_ from page 190 (section 2.1.2 and 2.1.3)
+- `Toradex Wiki`_
+
+
+At compilation time you have to choose which RAM will be used. This configuration is
+done in the file ``boards/arm/colibri_imx7d_m4/colibri_imx7d_m4.dts`` with "zephyr,flash" (when CONFIG_XIP=y) and "zephyr,sram"
+properties. The available configurations are:
+
+"zephyr,flash"
+- &ddr_code
+- &tcml_code
+- &ocram_code
+- &ocram_s_code
+- &ocram_pxp_code
+- &ocram_epdc_code
+
+"zephyr,sram"
+- &ddr_sys
+- &tcmu_sys
+- &ocram_sys
+- &ocram_s_sys
+- &ocram_pxp_sys
+- &ocram_epdc_sys
+
+
+Below you will find the instructions to load and run Zephyr on M4 from A7 using u-boot.
+
+Copy the compiled zephyr.bin to the first EXT partition of the SD card and plug into the
+board. Power it up and stop the u-boot execution.
+Set the u-boot environment variables and run the zephyr.bin from the appropriated memory
+configured in the Zephyr compilation:
+
+.. code-block:: console
+  setenv bootm4 'ext2load mmc 0:1 $m4addr $m4fw && dcache flush && bootaux $m4addr'
+  # TCML
+  setenv m4tcml 'setenv m4fw zephyr.bin; setenv m4addr 0x007F8000'
+  setenv bootm4tcml 'run m4tcml && run bootm4'
+  run bootm4tcml
+  # TCMU
+  setenv m4tcmu 'setenv m4fw zephyr.bin; setenv m4addr 0x00800000'
+  setenv bootm4tcmu 'run m4tcmu && run bootm4'
+  run bootm4tcmu
+  # OCRAM
+  setenv m4ocram 'setenv m4fw zephyr.bin; setenv m4addr 0x00900000'
+  setenv bootm4ocram 'run m4ocram && run bootm4'
+  run bootm4ocram
+  # OCRAM_S
+  setenv m4ocrams 'setenv m4fw zephyr.bin; setenv m4addr 0x00180000'
+  setenv bootm4ocrams 'run m4ocrams && run bootm4'
+  run bootm4ocrams
+  # DDR
+  setenv m4ddr 'setenv m4fw zephyr.bin; setenv m4addr 0x80000000'
+  setenv bootm4ddr 'run m4ddr && run bootm4'
+  run bootm4ddr
+
+
+Debugging
+=========
+
+Download and install `J-Link Tools`_ and `NXP iMX7D Connect CortexM4.JLinkScript`_.
+
+To run Zephyr Binary using J-Link create the following script in order to
+get the Program Counter and Stack Pointer from zephry.bin.
+
+get-pc-sp.sh:
+.. code-block:: console
+  #!/bin/sh
```

Comment:
need a blank line after the ``.. code-block`` directive, and all the code block lines need to be indented to the "c" in "code-block"